### DISCUSSION TITLE
Pluto TX-RX test - misaligned packet received from the DMA

### DISCUSSION THREAD

**Discussion by mbound** [srsran/zynq_timestamping#42] [General]
Hi all! I have been testing the Zynq timestamping with an ADALM PLUTO SDR. 

Had to make some minor tweaks to the txrx test script from the app note to make it run, namely correct the folder name from `bin_test` to `bin_app`, reducing the gain and of course set the right IP address.
The transmission seems to work but I get a bunch of notices regarding misaligned packets (see below).  Is this expected?

```
mbound@arrakeen:~/SDR/PLUTO/timestamp/zynq_timestamping/app/scripts$ ./run_txrx_plutosdr.sh 
++ pwd
+ sudo LD_LIBRARY_PATH=/home/mbound/SDR/PLUTO/timestamp/zynq_timestamping/app/scripts/../bin_app nice -20 ../bin_app/txrx_test -f 2400000000 -a n_prb=6,context=ip:192.168.2.1 -p 6 -g 10 -G 10 -o test_txrx_pluto.bin
Opening RF device...
Active RF plugins: libsrsran_rf_iio.so
Inactive RF plugins:
Supported RF device list: iio
Trying to open RF device 'iio'
CH0 n_prb=6
CH0 context=ip:192.168.2.1
failed to create the rf_port with A_BALENCED
failed to create the rf_port with A
RF device 'iio' successfully opened
Subframe len:   1920 samples
Time advance:   0.000000 us
Set TX/RX rate: 1.92 MHz
Set RX gain:    10.0 dB
Set TX gain:    10.0 dB
Set TX/RX freq: 2400.00 MHz
Rx subframe 0
misaligned packet received from the DMA
Transmitting Signal
Rx subframe 1
misaligned packet received from the DMA
Rx subframe 2
misaligned packet received from the DMA
Rx subframe 3
misaligned packet received from the DMA
Transmitting Signal
Rx subframe 4
misaligned packet received from the DMA
Rx subframe 5
misaligned packet received from the DMA
Rx subframe 6
misaligned packet received from the DMA
Rx subframe 7
misaligned packet received from the DMA
Rx subframe 8
misaligned packet received from the DMA
Transmitting Signal
Rx subframe 9
misaligned packet received from the DMA
Rx subframe 10
misaligned packet received from the DMA
Rx subframe 11
misaligned packet received from the DMA
Rx subframe 12
misaligned packet received from the DMA
Rx subframe 13
misaligned packet received from the DMA
Rx subframe 14
misaligned packet received from the DMA
Rx subframe 15
misaligned packet received from the DMA
Rx subframe 16
misaligned packet received from the DMA
Rx subframe 17
misaligned packet received from the DMA
Rx subframe 18
misaligned packet received from the DMA
Rx subframe 19
misaligned packet received from the DMA
misaligned packet received from the DMA
Done

```

**Comment by sauka** [comment]
Thanks for your feedback @mbound 

No, this is not an expected behaviour. Could you make sure the bitstream with timestamping support was correctly loaded? From the log above it looks like plutoSDR was programmed with a default bitstream 

(basically `misaligned packet` is reported on every receive() call, meaning that sw can't find a header with the sync words and timestamp in the received data).

**Reply by mbound** [reply]
Thanks, I was using the Makefile from the main repository, I re-ran the commands one by one from the app note and now it seems to work.
I've noted that upon disconnection of the Pluto, the firmware resets to the original one and the procedure has to be repeated, is there a way to make this permanent?

```
+ sudo LD_LIBRARY_PATH=/home/mbound/SDR/PLUTO/timestamp/zynq_timestamping/app/scripts/../bin_app nice -20 ../bin_app/txrx_test -f 2400000000 -a n_prb=6,context=ip:192.168.2.1 -p 6 -g 10 -G 10 -o test_txrx_pluto.bin
Opening RF device...
Active RF plugins: libsrsran_rf_iio.so
Inactive RF plugins:
Supported RF device list: iio
Trying to open RF device 'iio'
CH0 n_prb=6
CH0 context=ip:192.168.2.1
failed to create the rf_port with A_BALENCED
failed to create the rf_port with A
RF device 'iio' successfully opened
Subframe len:   1920 samples
Time advance:   0.000000 us
Set TX/RX rate: 1.92 MHz
Set RX gain:    10.0 dB
Set TX gain:    10.0 dB
Set TX/RX freq: 2400.00 MHz
Rx subframe 0
Transmitting Signal
Rx subframe 1
Rx subframe 2
Rx subframe 3
Transmitting Signal
Rx subframe 4
Rx subframe 5
Rx subframe 6
Rx subframe 7
Rx subframe 8
Transmitting Signal
Rx subframe 9
Rx subframe 10
Rx subframe 11
Rx subframe 12
Rx subframe 13
Rx subframe 14
Rx subframe 15
Rx subframe 16
Rx subframe 17
Rx subframe 18
Rx subframe 19
Done
```

**Reply by mbound** [reply]
However, if I run this in non-loopback mode and with another SDR monitoring the spectrum, I don't see anything being transmitted other than a DC leak.

**Reply by sauka** [reply]
Thanks! We need to check official Pluto documentation on how to change the behaviour on power-cycling. Your initial problem might be because of the issue described in #44 , we will have a look.
I see your messages in this and another thread about TX path being silent when Pluto is used as an RF frontend, I hope next week I have an opportunity to test it in HW and see what is the problem.

Thanks for your feedback!

**Reply by mbound** [reply]
Thanks, in the meantime I'll do some more experimenting and try to see if there's some useful feedback on the Pluto.  
Any suggestion on where I could start looking would be great :-)

EDIT: are you targeting a specific base firmware version? I am running 0.30 right now. 

Regarding #44, the order of commands in the Makefile and the app note is the same, and it seems to work for me at least when running commands manually one by one, so I am not sure what the prior issue might have been.
